src/main/java/adris/altoclef/ConfigState.java

```diff
@@ -179,31 +195,33 @@ public void allowShears(boolean allow) {
 
     /// Stack management
     public void push() {
-        if (_states.empty()) {
-            _states.push(new State());
+        if (states.empty()) {
+            states.push(new State());
         } else {
             // Make copy and push that
-            _states.push(new State(current()));
+            states.push(new State(current()));
         }
     }
+
     public void pop() {
-        if (_states.empty()) {
+        if (states.empty()) {
             Debug.logError("State stack is empty. This shouldn't be happening.");
             return;
         }
-        State s = _states.pop();
+        State s = states.pop();
         s.applyState();
     }
 
     private State current() {
-        if (_states.empty()) {
+        if (states.empty()) {
             Debug.logError("STATE EMPTY, UNEMPTIED!");
             push();
         }
-        return _states.peek();
+        return states.peek();
     }
 
-    class State {
+    public class State {
+        // TODO: 2021-05-22 public bad
```

Comment:
Valid, but perhaps not all. I allowed public variables here because State is only used in a private context. Making this private would result in each value having a getter and setter, and this might make this class bloated IMO.

Imagine this: I want to add a new value. Say, to force disable sprinting. How much work would we have to put in to add that feature? Could ConfigState.java be designed to make adding this feature easier?

Maybe it's not that big of a deal tbh, but still.

Reply:
if a variable is not `static final`, it should not be `public`. Even in private contexts.

Reply:
tbh, your whole "config state" class is kinda weird.

Just.. read a json file with the saved settings lol

Reply:
I agree that it's weird but it's never stored in a json file. There's a reason why "Settings.java" and "ConfigState.java" are in separate files, because they are totally different from each other.

Settings is: "The user wants to change how the bot acts globally"
ConfigState is: "The programmer wants to change how the bot acts for the duration of one task during runtime"

Reply:
then it should 100% be renamed